clean_project/construct_dict_test.py

```diff
@@ -0,0 +1,26 @@
+# -*- coding: utf-8 -*-
+import json
+import tempfile
+import unittest
+
+from construct_dict import construct_dictionary
+
+
+class ConstructDictTest(unittest.TestCase):
+
+    def test_empty_yaml_file(self):
+        with tempfile.NamedTemporaryFile(suffix="JSON") as json_file:
+            self.assertEqual("Empty yaml file", construct_dictionary('test_data/config_empty_file.yaml', json_file.name))
+
+    def test_full_dictionary(self):
+        with tempfile.NamedTemporaryFile(suffix="JSON") as json_file:
```

Comment:
i think this would be better and cleaner if you:

1. Created the file
2. Built the index
3. Open the file for json loading and assert
4. Delete the file before the end of the test.

here you are opening a file for reading, then adding content to it, then actually reading, which may be functionally fine but it looks dirty and not obvious